src/host_capabilities/verification.rs

```diff
@@ -14,36 +24,42 @@ pub fn verify_image(image: &str, config: Config) -> Result<bool> {
     let msg = serde_json::to_vec(&req)
         .map_err(|e| anyhow!("error serializing the validation request: {}", e))?;
 
-    let response_raw = wapc_invoke_verify(&msg)?;
+    let response_raw = wapc_verify_fn(&msg)?;
 
-    let verified: bool = serde_json::from_slice(&response_raw)
-        .map_err(|e| anyhow!("Cannot decode verification response: {}", e))?;
+    let verified = *response_raw
+        .first()
+        .ok_or_else(|| anyhow!("error parsing the callback response"))?
+        != 0;
```

Comment:
Right now all the data exchanged via waPC is JSON encoded. We have to go back to the original implementation on the left, the one doing a `serde_json::from_slice`

Reply:
Actually we always [return true](https://github.com/kubewarden/policy-evaluator/pull/112/files#diff-8a3183dbac3fcc753fcf63d3b461057d35c455d6213bfb60b5c1d0c68cd5833bR43) is verification was successful or an error if it failed. Can we just remove the boolean and return an error if the result contains an error or true otherwise? I believe there is no point in having a boolean if we never check if it is false.

Reply:
I agree with Raul. Or do we return a bool because we need to return some bytes on the response payload?